File: pruebaFront/src/app/models/producto.ts
export class Producto {
    id?: number;
    creation_date:Date;
    description: string
    expiry_date:Date;
    name:string;
    price:number

    constructor(id?: number, creation_date?: Date, description?: string, expiry_date?: Date, name?: string, price?: number){
        this.id = id || 0;
        this.creation_date = creation_date || new Date;
        this.description = description || '';
        this.expiry_date = expiry_date || new Date;
        this.name = name || '';
        this.price =price || 0;
    }
}

File: prueba/src/main/java/com/prueba/prueba/entity/Producto.java
package com.prueba.prueba.entity;

import java.util.Date;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class Producto {
    @Id
    private int id;
    private Date creation_date;
    private String description;
    private Date expiry_date;
    private String name;
    private int price;

    public Producto() {

    }

    public Producto(int id, Date creation_date, String description, Date expiry_date, String name, int price) {
        this.id = id;
        this.creation_date = creation_date;
        this.description = description;
        this.expiry_date = expiry_date;
        this.name = name;
        this.price = price;
    }

    public int getId() {
        return this.id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getCreation_date() {
        return this.creation_date;
    }

    public void setCreation_date(Date creation_date) {
        this.creation_date = creation_date;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getExpiry_date() {
        return this.expiry_date;
    }

    public void setExpiry_date(Date expiry_date) {
        this.expiry_date = expiry_date;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPrice() {
        return this.price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

}
